### DISCUSSION TITLE
Multi-UE and performance testing

### DISCUSSION THREAD

**Discussion by Borjis131** [HewlettPackard/PacketRusher#141] [Q&A] [answered]
Hi @linouxis9,

I have seen that I can use the `ue` option to request PacketRusher to create a gNB and UE with a tunnel and then use the tunnel to send some traffic to the 5G Core using the VRF created. That works great.

I also have seen that I am able to create multiple UEs using the `multi-ue` option, in order to have a number of UEs connecting to one gNB. That works great for testing the connection. But checking the `multi-ue` option, if I want to use the tunnel of these UEs I need to use the options `--tunnel` and `--dedicatedGnb`. That forces me to have configured multiple IP addresses (one per gNB/UE pair) in the machine that I am running PacketRusher and that is a bit limiting.

**I want to find the correct way to spawn a high number of UEs (20 per example) and have them performing some traffic to the 5G Core. (preferably not using a gNB per each UE).**

**Is the `multi-ue` with the tunnel options the only way? Is this something for the `custom-scenario` feature?**

I know PacketRusher is still under development and I am probably asking for a more advanced feature, but I ask only to know if there is something that I am missing.

Thanks a lot!

**Comment by linouxis9** [comment, marked as answer]
Hi @Borjis131,

I hope you are doing well!
I've discussed this multiple times in the discussions / issues on Github (for ex. https://github.com/HewlettPackard/PacketRusher/issues/111#issuecomment-2049126443 https://github.com/HewlettPackard/PacketRusher/issues/88 https://github.com/HewlettPackard/PacketRusher/discussions/123) and indeed, this is a big limitation on PacketRusher. As you rightly pointed out, for now, the proper way of running PacketRusher with multi-ue that need a tunnel, is to create a gNodeB per UE using `--dedicatedGnb`.

Sadly, I had not the time to work more on this side, but this PR is supposed to fix this issue once done, but it's still very WIP: https://github.com/HewlettPackard/PacketRusher/pull/47

Sadly, that's all I have for you for now :(

Sorry and thanks for your interest on using PacketRusher!
Valentin

**Reply by Borjis131** [reply]
I see, sorry for opening a new discussion with the same topic then!

I also see no commits since a couple of months, are you working on a big feature or is that PacketRusher is no longer your focus? I still don't know how to treat it because I see you as the main developer but it is also on the HewlettPackard organization hahaha

Hope that everything is going well for you 🫂

**Reply by linouxis9** [reply]
No worries at all @Borjis131!

This is a project I started while working at HPE on company's time, that's why it's under the HewlettPackard organization, but nowadays nobody from HPE work on the project. Actually, I've left HPE at almost the same time the project open sourced and most of the work here has been done on my free time since (that's why you can see in the code a mix of copyright under HPE's and my name). I would have preferred to have this project rightfully under my alias :(

Everything is going well for me! I've been working on exciting new 5G projects, but I hope to go back to PacketRusher at some point as well.

**Reply by Borjis131** [reply]
Ohhhh I see. Well, you can always fork the project on your user and continue the work from there, there is no issue with that!

So great that everything is going well for you, I hope we get to see those exciting 5G projects you mention. You have one follower here I will continue to take a look at your projects if you post them 😄  

You did a great work with PacketRusher it is a shame that HPE is not putting some resources on it
